doc/howto/storage_resize.md

```diff
@@ -0,0 +1,66 @@
+---
+discourse: 1333
+---
+
+# How to resize storage
+
+If you need more storage, you can increase the size of your storage pool or your storage volume.
+In some cases, it is also possible to reduce the size of a storage volume.
+
+## Grow a storage pool
+
+To increase the size of a storage pool, follow these general steps:
+
+1. Grow the size of the storage on disk.
+1. Let the file system know of the size change.
+1. Update the pool size configuration in LXD.
+
+See the specific commands for different storage drivers below.
+
+````{tabs}
+
+```{group-tab} Btrfs
+
+Enter the following commands to grow a loop-backed Btrfs pool by 5 G:
+
+    sudo truncate -s +5G <LXD_lib_dir>/disks/<pool_name>.img
+    sudo losetup -c <loop_device>
+    sudo btrfs filesystem resize max <LXD_lib_dir>/storage-pools/<pool_name>/
+    lxc storage set <pool_name> size <new_size>
+
+Replace `<LXD_lib_dir>` with `/var/snap/lxd/common/mntns/var/snap/lxd/common/lxd/` if you are using the snap, or with `/var/lib/lxd/` otherwise.
+
+Replace `loop_device` with the mounted loop device that is associated with the storage pool image (for example, `/dev/loop8`).
+To find all mounted loop devices, enter `losetup -l`.
+```
+```{group-tab} ZFS
+
+Enter the following commands to grow a loop-backed ZFS pool by 5 G:
+
+    sudo truncate -s +5G <LXD_lib_dir>/disks/<pool_name>.img
+    sudo zpool set autoexpand=on <pool_name>
+    sudo zpool online -e <pool_name> <device_ID>
+    sudo zpool set autoexpand=off <pool_name>
+    lxc storage set <pool_name> size <new_size>
+
+Replace `<LXD_lib_dir>` with `/var/snap/lxd/common/lxd/` if you are using the snap, or with `/var/lib/lxd/` otherwise.
+
+Replace `<device_ID>` with the ID of the ZFS device.
+Enter `sudo zpool status -vg <pool_name>` to find the ID.
+```
+
```

Comment:
Can we add steps for LVM as well?

Reply:
Will take a look at getting steps identified.

Reply:
For LVM both LVM thin and non-thin pools:

```
lxc storage create lvm2 lvm size=10GiB
sudo losetup -j /var/lib/lxd/disks/lvm2.img # Find associated loop device for disk image.
/dev/loop1: []: (/var/lib/lxd/disks/lvm2.img) 
sudo truncate /var/lib/lxd/disks/lvm2.img -s 20GiB # Expand backing loop file.
sudo losetup -c /dev/loop1 # Refresh loop device size.
sudo pvresize /dev/loop1 # Expand physical volume.
sudo pvs /dev/loop1 # Check physical volume resized.
  PV         VG   Fmt  Attr PSize   PFree 
  /dev/loop1 lvm2 lvm2 a--  <20.00g 10.00g 
sudo vgs lvm2 # Check volume group resized.
  VG   #PV #LV #SN Attr   VSize   VFree 
  lvm2   1   1   0 wz--n- <20.00g 10.00g
```

Finally, just for LVM thin pools, expand the thinpool logical volume:

```
sudo lvextend lvm2/LXDThinPool -l+100%FREE
sudo lvs lvm2/LXDThinPool # Check thinpool LV is resized.
  LV          VG   Attr       LSize   Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  LXDThinPool lvm2 twi-a-tz-- <19.96g             0.00   10.57                          
```

Reply:
Thanks @tomponline !
Added now.